jazkarta/shop/browser/checkout/stripe.py

```diff
@@ -42,34 +42,45 @@ def handle_submit(self):
         contact_info = PersistentMapping()
         for f in ('first_name', 'last_name', 'email', 'phone', 'address',
                   'city', 'state', 'zip', 'country'):
-            contact_info[f] = self.request.form[f]
-
-        method = 'Online Payment'
-        if self.is_superuser():
-            method = self.request.form.get('method', 'None')
-            if method not in ('Online Payment', 'Check', 'Cash', 'None'):
-                raise Forbidden('Invalid payment method: %s' % method)
-            if amount and (method == 'None'):
-                raise Forbidden('Invalid payment method: None')
-
-        charge_result = {'success': True}
-        if amount and method == 'Online Payment':
-            if 'stripeToken' not in self.request.form:
-                self.error = 'Unable to process payment. Please try again.'
-                return
-            card_token = self.request.form['stripeToken']
-
-            try:
-                charge_result = process_interactive_payment(
-                    self.cart, card_token, contact_info)
-                charge_result['success'] = True
-            except PaymentProcessingException as e:
-                charge_result = {
-                    'success': False,
-                    'err_msg': e.message,
-                    'err_code': getattr(e, 'code', None),
-                }
-                self.error = e.message
+            if f in self.request.form: # for digital products only email address is currently required in /checkout
+                contact_info[f] = self.request.form[f]
+            else:
+                contact_info[f] = None
+
+        if 'nocharge' not in self.request.form:
```

Comment:
Relying on a value from the request here seems dangerous. I think we should check the server-side cart amount to make this determination.

Reply:
I think checking both `not amount` and the request value should suffice here, a little redundant but safe.